### PR TITLE
Add missing exception javadoc/tests for some null arguments

### PR DESCRIPTION
**Pull request by dvmarcilio**
Hello! I added some missing javadoc for null arguments and wrote tests when they didn't exist.
I tried to keep the diffs as small as possible and the sentences consistent.
I will be happy to change anything if needed. Thanks.

### CODE REVIEW

**Comment by garydgregory** [comment]
Hi @dvmarcilio & thank you for your PR. Let' hold off on this one. I want to normalize on using Java's Objects#requireNonNull().

**Review by garydgregory — CHANGES_REQUESTED** [review]
Hello @dvmarcilio 
Thank you for your patience and PR. Please rebase on git master and review your PR with my comment in mind. 
TY!

**Comment by dvmarcilio** [comment]
Hi @garydgregory. 
Thanks for pinging me. All should be fine now.

File: src/main/java/org/apache/commons/lang3/reflect/TypeUtils.java [hunk 1]
```diff
@@ -1607,6 +1609,8 @@ private static String parameterizedTypeToString(final ParameterizedType paramete
      * @param rawClass the raw class to create a parameterized type instance for
      * @param typeVariableMap the map used for parameterization
      * @return {@link ParameterizedType}
+     * @throws NullPointerException if either {@code rawClass} or {@code typeVariable}
```
Comment (kinow) [inline src/main/java/org/apache/commons/lang3/reflect/TypeUtils.java:1612]: `typeVariableMap` & `typeVariableMap`?

**Review by kinow — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/commons/lang3/reflect/TypeUtils.java (hunk 1 above)
Comment (dvmarcilio) [reply src/main/java/org/apache/commons/lang3/reflect/TypeUtils.java]: Oops. Thanks for finding it. Fixed it!
